### PR TITLE
It3 gameteam seasonid refactor

### LINKED ISSUE #78: GameTeam Seasonid Refactor
I just realized that the game_id always starts with the start year of the season:

game_id: **2015**030142 , season_id: **2015**2016
game_id: **2016**030172, season_id: **2016**2017

This means we can generate the season_id attribute in the GameTeam data without having to pull in data from Game and thus fully eliminate nested iteration.

### PR DESCRIPTION
**Pull request by AngelaGuardia**
**Changes made**
- Changed the way we add the season_id attribute to the GeamTeam objects by generating it directly from the GameTeam data
- Deleted/refatored all code related to the old way of adding season_id to GameTeam

### CODE REVIEW

File: test/game_team_test.rb [hunk 1]
```diff
@@ -13,8 +13,10 @@ def setup
                                 :head_coach=>"John Tortorella",
                                 :goals=>2,
                                 :shots=>8,
-                                :tackles=>44
+                                :tackles=>44,
```
Comment (leahriffell) [inline test/game_team_test.rb]: Did you mean to add this comma here?

**Review by leahriffell — APPROVED** [review]
Great catch on this! Thank you! Much better than having to iterate through.

File: lib/game_team_collection.rb [hunk 2]
```diff
@@ -5,22 +5,21 @@ class GameTeamCollection
...
   def create_game_teams
     CSV.foreach(@game_team_path, headers: true, header_converters: :symbol) do |row|
       game_team = GameTeam.new(row.to_h)
       @all_game_teams << game_team
-      @all_gameids_per_season.each do |season, game_ids|
-        if game_ids.include?(game_team.game_id)
-          game_team.season = season
-        end
-      end
+      game_team.season = generate_season(game_team.game_id)
```
Comment (sarynm12) [inline lib/game_team_collection.rb:18]: Nice :)

**Review by sarynm12 — APPROVED** [review]

File: test/game_team_test.rb (hunk 1 above)
Comment (AngelaGuardia) [reply test/game_team_test.rb]: Nice catch @leahriffell! Just corrected it.

**Review by hopesgit — APPROVED** [review]

**Pull request merged**